pkg/declarative/v2/helm.go

```diff
@@ -0,0 +1,160 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-project/module-manager/pkg/types"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/kube"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+const (
+	ConditionTypeHelmCRDs               ConditionType   = "HelmCRDs"
+	ConditionReasonHelmCRDsAreAvailable ConditionReason = "HelmCRDsAvailable"
+)
+
+func NewHelmRenderer(
+	spec *Spec,
+	clnt Client,
+	options *Options,
+) Renderer {
+	valuesAsMap, ok := spec.Values.(map[string]any)
+	if !ok {
+		valuesAsMap = map[string]any{}
+	}
+	return &Helm{
+		recorder:   options.EventRecorder,
+		chartPath:  spec.Path,
+		values:     valuesAsMap,
+		clnt:       clnt,
+		crdChecker: NewHelmReadyCheck(clnt),
+	}
+}
+
+type Helm struct {
+	recorder record.EventRecorder
+	clnt     Client
+
+	chartPath string
+	values    map[string]any
+
+	chart *chart.Chart
+	crds  kube.ResourceList
+
+	crdChecker ReadyCheck
+}
+
+func (h *Helm) prerequisiteCondition(object metav1.Object) metav1.Condition {
+	return metav1.Condition{
+		Type:               string(ConditionTypeHelmCRDs),
+		Reason:             string(ConditionReasonHelmCRDsAreAvailable),
+		Status:             metav1.ConditionFalse,
+		Message:            "CustomResourceDefinitions from chart 'crds' folder are installed and ready for use",
+		ObservedGeneration: object.GetGeneration(),
+	}
+}
+
+func (h *Helm) Initialize(obj Object) error {
+	status := obj.GetStatus()
+
+	prerequisiteExists := meta.FindStatusCondition(status.Conditions, h.prerequisiteCondition(obj).Type) != nil
+	if !prerequisiteExists {
+		status := obj.GetStatus()
+		meta.SetStatusCondition(&status.Conditions, h.prerequisiteCondition(obj))
+		obj.SetStatus(status)
```

Comment:
The `status` haven't changed, why do we need to set same status, and what's the different between status in line 63 and 67, does it changes?